spec/db/db_sqlite_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'db/db_sqlite'
+
+describe DBSQLite do
+
+  let (:sqlite)  {SQLite3::Database.new ":memory:"}
+  let (:db)      {described_class.new(sqlite, "tablename", ["name", "address"])}
+  let (:contact) {{"name" => "a name", "address" => "an address"}}
+
+  it "returns an empty array if database is empty" do
+    expect(db.all).to eq([])
+  end
+
+  it "adds one contact to the database" do
+    db.add(contact)
+    expect(db.all).to eq([contact_with_id(1)])
+  end
+
+  it "retrieves all contacts" do
+    db.add(contact)
+    db.add(contact)
+    db.add(contact)
+    contacts = [contact_with_id(1), contact_with_id(2), contact_with_id(3)]
+    expect(db.all).to eq(contacts)
+    expect(db.size).to eq(3)
+  end
+
+  it "gets an empty contact if there are no contacts in the db" do
+    expect(db.at(0)).to eq({})
+  end
+
+  it "gets an empty contact if the index is not valid" do
+    db.add(contact)
+    expect(db.size).to eq(1)
+    expect(db.at(5)).to eq({})
+  end
+
+  it "gets one contact if there are contacts and the index is valid" do
+    db.add(contact)
+    expect(db.at(0)).to eq(contact_with_id(1))
+  end
+
+  it "updates a contact if a contact with the same id is passed" do
+    db.add(contact)
+    contact["name"] = "new name"
+    db.update(contact_with_id(1))
+    expect(db.size).to eq(1)
+    expect(db.at(0)["name"]).to eq("new name")
+  end
+
+  it "finds one contact given an exact term" do
+    db.add(contact)
+
+    contact2 = contact
+    contact2["address"] = "address2"
+    db.add(contact2)
+
+    contact3 = contact
+    contact3["address"] = "address3"
+    db.add(contact3)
```

Comment:
I didn't add the helper method because I didn't have three repetitions.

:boom:

It's added now :+1: